TestModels/ExternV2/README.md

```diff
@@ -0,0 +1,67 @@
+# SimpleExternV2
+
+This project tests implementing a [dafny extern](https://homepage.cs.uiowa.edu/~tinelli/classes/181/Papers/dafny-reference.pdf#15) using the "V2 Extern" system.
+This models and its shapes are based on to operations in the `Extern` TestModel.
+
+The differences between these two TestModels are:
+* V2 suffixes are appended to classes and operations
+* The NET implementation of the `simple.dafnyexternv2.internaldafny` extern contains an extern method that is not present in other languages (as a demonstration; see below)
+* The `ExternV2Constructor` is moved to `simple.dafnyexternv2.internaldafny.ExternV2Constructor` 
+
+This TestModel also demonstrates a "V2" extern system.
+
+The "V1" extern system applies an `{:extern "..."}` attribute to a module to signal an extern.
+This worked for .NET and Java, where this extern string was valid in both languages.
+However, this does not work for Python and Go, where this string is not applicable.
+
+The "V2" extern system is intended to allow per-language extern strings.
+This system uses the `replaceable` and `replaces` keywords
+in conjunction with the `outer-module` compile flag
+to allow a module to have idiomatic extern names in each target language.
```

Comment:
Issue/Blocking/Question: I thought `outer-module` did not work,
and we were removing it for just `replaceable`?

Reply:
That's inaccurate; we want to use both. This TestModel only implements `replaceable` at the moment, but will implement `outer-module` (or its replacement) once that's ready.